astropy/utils/tests/test_data.py

```diff
@@ -372,3 +376,37 @@ def test_is_url_in_cache():
 
     download_file(TESTURL, cache=True, show_progress=False)
     assert is_url_in_cache(TESTURL)
+
+
+@contextlib.contextmanager
+def TemporaryTempDir():
+    """Temporarily redirect all temporary files to a temporary directory"""
+    tempdir = tempfile.mkdtemp()
+    old_tempdir = tempfile.tempdir
+    tempfile.tempdir = tempdir
+    try:
+        yield tempdir
+    finally:
+        tempfile.tempdir = old_tempdir
+        shutil.rmtree(tempdir)
+
+
+def test_get_readable_fileobj_cleans_up_temporary_files():
+    """checks that get_readable_fileobj leaves no temporary files behind"""
+    from ..data import get_pkg_data_filename, get_readable_fileobj
```

Comment:
On Sep 30, 2015, at 12:14 PM, Erik Bray <notifications@github.com<mailto:notifications@github.com>> wrote:


In astropy/utils/tests/test_data.py<https://github.com/astropy/astropy/pull/4198#discussion_r40814887>:

> +@contextlib.contextmanager
> +def TemporaryTempDir():
> +    """Temporarily redirect all temporary files to a temporary directory"""
> +    tempdir = tempfile.mkdtemp()
> +    old_tempdir = tempfile.tempdir
> +    tempfile.tempdir = tempdir
> +    try:
> +        yield tempdir
> +    finally:
> +        tempfile.tempdir = old_tempdir
> +        shutil.rmtree(tempdir)
> +
> +
> +def test_get_readable_fileobj_cleans_up_temporary_files():
> +    """checks that get_readable_fileobj leaves no temporary files behind"""
> +    from ..data import get_pkg_data_filename, get_readable_fileobj


The in-line imports aren't necessary either. You can just import these at the top of the file (if there are any tests already that do it, they're demonstrating an anti-pattern)

—
Reply to this email directly or view it on GitHub<https://github.com/astropy/astropy/pull/4198/files#r40814887>.

Fixed via force push.